exporter/awsemfexporter/datapoint.go

```diff
@@ -213,42 +235,122 @@ func (dps histogramDataPointSlice) IsStaleNaNInf(i int) (bool, pcommon.Map) {
 }
 
 // CalculateDeltaDatapoints retrieves the ExponentialHistogramDataPoint at the given index.
+// As CloudWatch EMF logs allows in maximum of 100 target members, the exponential histogram metric are split into two data points if the total buckets exceed 100,
+// the first data point contains the first 100 buckets, while the second data point includes the remaining buckets.
+// Re-calculated Min, Max, Sum, Count for each split:
+// 1. First split datapoint:
+//   - Max: From original metric.
+//   - Min: Last bucket’s bucketBegin in the first split.
+//   - Sum: From original metric.
+//   - Count: Calculated from the first split buckets.
+//
+// 2. Second split datapoint:
+//   - Max: First bucket’s bucketEnd in the second split.
+//   - Min: From original metric.
+//   - Sum: 0.
+//   - Count: Overall count - first split count.
 func (dps exponentialHistogramDataPointSlice) CalculateDeltaDatapoints(idx int, _ string, _ bool, _ *emfCalculators) ([]dataPoint, bool) {
```

Comment:
I know for our usecase, we only expect a max of 160 buckets, but if we get more than 200, this still won't work. Is there a reason why we're restricting ourselves to 2 datapoint splits? Can't we support N datapoint splits?

Reply:
You are correct, I was checking the otel upstream Exponential Bucket Histogram Aggregation has [Maximum number of buckets 160](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#base2-exponential-bucket-histogram-aggregation). But it turns out the 160 is the maximum number in each of the positive and negative ranges, so the total number can be more than 300. I have changed to support N datapoint splits.